### PR TITLE
feat(aws-cli): 🧑‍💻 add aws-cli completion

### PR DESCRIPTION
**Pull request by rubensa**
**aws-cli** supports **bash** and **zsh** command completion.

### CODE REVIEW

**Comment by rubensa** [comment]
Branch re-based from main.

File: src/aws-cli/aws_zsh_completer.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Source this file to activate auto completion for zsh using the bash
```
Comment (samruddhikhandale) [inline src/aws-cli/aws_zsh_completer.sh:1]: Is this file published anywhere by a trusted source? Can you point me to it?

Comment (rubensa) [reply src/aws-cli/aws_zsh_completer.sh:1]: It's in [aws-cli sources](https://github.com/aws/aws-cli/blob/v2/bin/aws_zsh_completer.sh) but it's not included with the installer.

**Comment by rubensa** [comment]
Branch re-based from main.

File: src/aws-cli/aws_zsh_completer.sh (hunk 1 above)
Comment (samruddhikhandale) [reply src/aws-cli/aws_zsh_completer.sh:1]: Thanks! As we are copying this script file, it's very likely to get stale in long run. 

Is it possible to add an action similar to https://github.com/devcontainers/features/blob/main/.github/workflows/update-dotnet-install-script.yml which syncs this file? Thanks!

File: src/aws-cli/aws_bash_completer [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Typically that would be added under one of the following paths:
+# - /etc/bash_completion.d
+# - /usr/local/etc/bash_completion.d
+# - /usr/share/bash-completion/completions
```
Comment (samruddhikhandale) [inline src/aws-cli/aws_bash_completer:4]: Same comment here.

File: src/aws-cli/aws_zsh_completer.sh (hunk 1 above)
Comment (samruddhikhandale) [inline src/aws-cli/aws_zsh_completer.sh:1]: Also, let's move these scripts under a new `vendor-scripts` folder and add a README similar to https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/vendor/README.md#important-note

File: src/aws-cli/install.sh [hunk 3]
```diff
@@ -114,6 +113,19 @@ install() {
...
+    # kubectl bash completion
+    ls -la ./aws_bash_completer
+    mkdir -p /etc/bash_completion.d
+    ls -la /etc/bash_completion.d
+    cp ./aws_bash_completer /etc/bash_completion.d/aws
+
+    # kubectl zsh completion
+    if [ -e "${USERHOME}/.oh-my-zsh" ]; then
+        mkdir -p "${USERHOME}/.oh-my-zsh/completions"
+        cp ./aws_zsh_completer.sh > "${USERHOME}/.oh-my-zsh/completions/_aws"
```
Comment (samruddhikhandale) [inline src/aws-cli/install.sh:125]:
```suggestion
        cp ./aws_zsh_completer.sh "${USERHOME}/.oh-my-zsh/completions/_aws"
```

File: test/aws-cli/checkBashCompletion.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+command=$1
+expected=$2
```
Comment (samruddhikhandale) [inline test/aws-cli/checkBashCompletion.sh:4]: Thanks for adding test coverage!

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: .github/workflows/update-aws-cli-completer-scripts.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          echo "Start."
+
+          # Configure git and Push updates
+          git config --global user.email github-actions@github.com
+          git config --global user.name github-actions
+          git config pull.rebase false
+
+          branch=automated-script-update-$GITHUB_RUN_ID
+          git checkout -b $branch
+          message='[Updates] Automated vendor completer scripts'
```
Comment (samruddhikhandale) [inline .github/workflows/update-aws-cli-completer-scripts.yml:35]:
```suggestion
          message='[Updates] Automated vendor 'aws-cli' completer scripts'
```

File: src/aws-cli/install.sh [hunk 6]
```diff
@@ -114,6 +113,19 @@ install() {
     unzip "${scriptZipFile}"
     ./aws/install
 
+    # kubectl bash completion
+    ls -la ./aws_bash_completer
+    mkdir -p /etc/bash_completion.d
+    ls -la /etc/bash_completion.d
```
Comment (samruddhikhandale) [inline src/aws-cli/install.sh:119]:
```suggestion
    mkdir -p /etc/bash_completion.d
```

File: test/aws-cli/checkBashCompletion.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (samruddhikhandale) [inline test/aws-cli/checkBashCompletion.sh:1]: Can you help fix the failing tests? Thank you!

File: .github/workflows/update-aws-cli-completer-scripts.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
+name: "Updates vendor 'aws_bash_completer' and 'aws_zsh_completer.sh' scripts"
```
Comment (samruddhikhandale) [inline .github/workflows/update-aws-cli-completer-scripts.yml:1]: Amazing 🎉

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

**Review by samruddhikhandale — APPROVED** [review]
Looks great, thank you ⚡

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**